specs/metered-guidance-errors.md

```diff
@@ -0,0 +1,111 @@
+[   API   ]  ┌ Passed through
+             └ GET https://graph.microsoft.com/v1.0/teams/*/channels/getAllMessages
+[ WARNING ]  ┌ This is a metered API.
+             │ More info at https://learn.microsoft.com/en-us/graph/teams-licenses
+             └ GET https://graph.microsoft.com/v1.0/teams/*/channels/getAllMessages 
+```
+
+### Random Errors
+When registered, the plugin should register 3 new errors for the `GraphRandomErrorPlugin` (The list is coming from https://learn.microsoft.com/en-us/graph/teams-licenses#payment-related-errors):
+
+| Error code | Scenario | Sample error message |
+|:-----------|:-----------|:-----------------|
+| 402 (Payment Required) | Passing `model=A` without a Microsoft E5 license |`...needs a valid license to access this API...`, `...tenant needs a valid license to access this API...`|
+| 402 (Payment Required) | Calling Patch API passing `model=B` |`...query parameter 'model' does not support value 'B' for this API. Use billing model 'A'...`|
+| 402 (Payment Required) | `Evaluation mode` capacity exceeded |`...evaluation mode capacity has been exceeded. Use a valid billing model...`|
+
+## Configuration file
+
+The plugin will have its own section in the `appSettings.json` configuration file where metered APIs will be listed. This file should have default values for all supported Microsoft Graph clouds.
+
+```json
+{
+  "name": "GraphMeteredGuidancePlugin",
+  "enabled": true,
+  "pluginPath": "GraphProxyPlugins\\msgraph-developer-proxy-plugins.dll",      
+  "meteredApis": [
+    {
+      "urlsToWatch": [
+        "https://graph.microsoft.com/v1.0/users/*/chats/getAllMessages",
+        "https://graph.microsoft.com/beta/users/*/chats/getAllMessages",
+      ],
+      "supportedPaymentModels": [
+        "model=A",
+        "model=B"
+      ],
+      "documentationUrl": "",
+      "billingInformation": "https://learn.microsoft.com/en-us/graph/teams-licenses",
+      "registrationForm": "https://aka.ms/teamsgraph/requestaccess"
+    },
+    {
+      "urlsToWatch": [
+        "https://graph.microsoft.com/v1.0/teams/*/channels/getAllMessages",
+        "https://graph.microsoft.com/beta/teams/*/channels/getAllMessages",
+      ],
+      "supportedPaymentModels": [
+        "model=A",
+        "model=B"
+      ],
+      "billingInformation": "https://learn.microsoft.com/en-us/graph/teams-licenses",
+      "registrationForm": "https://aka.ms/teamsgraph/requestaccess"
+    },
+    {
+      "urlsToWatch": [
+        "https://graph.microsoft.com/v1.0/teams/*/channels/*/messages/*",
+        "https://graph.microsoft.com/beta/teams/*/channels/*/messages/*",
+        "https://graph.microsoft.com/v1.0/teams/*/channels/*/messages/*/replies/*",
+        "https://graph.microsoft.com/beta/teams/*/channels/*/messages/*/replies/*",
+        "https://graph.microsoft.com/v1.0/chats/*/messages/{message-id}*",
+        "https://graph.microsoft.com/beta/chats/*/messages/{message-id}*",
+      ],
+      "httpVerbs": [

```

Comment:
This is an important implementation detail that we currently support only in mocks. We should mention it explicitly so that it's not lost in the implementation.